hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +85,964 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
+    ArrayHandle<int3> h_image(m_pdata->getImages(), access_location::host, access_mode::read);
+    const BoxDim& global_box = m_pdata->getGlobalBox();
+
+    ArrayHandle<Scalar3> h_linmom_accum(m_linmom_accum,
+                                        access_location::host,
+                                        access_mode::readwrite);
+    ArrayHandle<Scalar3> h_angmom_accum(m_angmom_accum,
+                                        access_location::host,
+                                        access_mode::readwrite);
+
+    unsigned int num_total = m_pdata->getN();
+    uint16_t seed = m_sysdef->getSeed();
+
+    // get random velocities and accumulate the resulting change in momentum
+    for (unsigned int idx = 0; idx < num_total; ++idx)
+        {
+        // get the index from the embedded group and check if in a rigid body
+        const unsigned int central_tag = h_body.data[idx];
+        if (central_tag >= MIN_FLOPPY)
+            {
+            continue;
+            }
+        const unsigned int central_idx = h_rtag.data[central_tag];
+        // if the central particle is not local, cannot read or write to it.
+        assert(central_idx != NOT_LOCAL);
+
+        // do not need to thermalize central particle
+        if (idx == central_idx)
+            {
+            continue;
+            }
+
+        Scalar mass_const = h_velocity.data[idx].w;
+        unsigned int tag = h_tag.data[idx];
+        // draw random velocities from normal distribution
+        hoomd::RandomGenerator rng(
+            hoomd::Seed(hoomd::RNGIdentifier::CollisionMethod, timestep, seed),
+            hoomd::Counter(tag));
+        hoomd::NormalDistribution<Scalar> gen(fast::sqrt(T_set / mass_const), 0.0);
+        Scalar3 vel;
+        gen(vel.x, vel.y, rng);
+        vel.z = gen(rng);
+        h_alt_vel.data[idx] = make_scalar4(vel.x, vel.y, vel.z, mass_const);
+
+        // add to net momentum
+        const Scalar3 linmom_change = mass_const * vel;
+        h_linmom_accum.data[central_idx] += linmom_change;
+
+        // get displacement
+        const Scalar4 postype_const = h_postype.data[idx];
+        const vec3<Scalar> pos_const(postype_const);
+        const int3 img_const = h_image.data[idx];
+
+        const Scalar4 postype_central = h_postype.data[central_idx];
+        const vec3<Scalar> pos_central(postype_central);
+        const int3 img_central = h_image.data[central_idx];
+
+        vec3<Scalar> displacement = pos_const - pos_central;
+        const int3 displacement_img = make_int3(img_const.x - img_central.x,
+                                                img_const.y - img_central.y,
+                                                img_const.z - img_central.z);
+        displacement = global_box.shift(displacement, displacement_img);
+
+        // add to net momentum
+        const vec3<Scalar> rand_vel(vel);
+        const vec3<Scalar> angmom_change = mass_const * cross(displacement, rand_vel);
```

Comment:
This is equivalent and reuses the multiplication, yes? If so, please replace anywhere this is repeated elsewhere too.
```suggestion
        const vec3<Scalar> angmom_change = cross(displacement, vec3<Scalar>(linmom_change));
```